src/Elliptic/Systems/Poisson/Equations.cpp

```diff
@@ -0,0 +1,163 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Elliptic/Systems/Poisson/Equations.hpp"
+
+#include <array>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"  // IWYU pragma: keep
+#include "Elliptic/Systems/Poisson/FirstOrderSystem.hpp"
+#include "Elliptic/Systems/Poisson/Tags.hpp"  // IWYU pragma: keep
+#include "NumericalAlgorithms/LinearOperators/Divergence.hpp"  // IWYU pragma: keep
+#include "NumericalAlgorithms/LinearOperators/PartialDerivatives.hpp"  // IWYU pragma: keep
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/TMPL.hpp"
+// IWYU pragma: no_forward_declare Tags::deriv
+// IWYU pragma: no_forward_declare Tags::div
+// IWYU pragma: no_forward_declare Variables
+// IWYU pragma: no_forward_declare Tensor
+
+namespace Poisson {
+
+template <size_t Dim>
+void ComputeFirstOrderOperatorAction<Dim>::apply(
+    const gsl::not_null<Scalar<DataVector>*> operator_for_field_source,
+    const gsl::not_null<tnsr::I<DataVector, Dim, Frame::Inertial>*>
+        operator_for_auxiliary_field_source,
+    const tnsr::i<DataVector, Dim, Frame::Inertial>& grad_field,
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& auxiliary_field,
+    const Mesh<Dim>& mesh,
+    const InverseJacobian<DataVector, Dim, Frame::Logical, Frame::Inertial>&
+        inverse_jacobian) noexcept {
+  auto div_vars = make_with_value<Variables<tmpl::list<AuxiliaryField<Dim>>>>(
+      auxiliary_field, 0.);
+  get<AuxiliaryField<Dim>>(div_vars) = auxiliary_field;
+  *operator_for_field_source = get<Tags::div<AuxiliaryField<Dim>>>(
+      divergence(div_vars, mesh, inverse_jacobian));
+  // Tensors don't support math operations yet, so we have to `get` the
+  // DataVector for the sign flip
+  get(*operator_for_field_source) *= -1.;
```

Comment:
So you're doing this instead of `get(*operator_for_field_source) = -1.0 * get(auxiliary_field;
  *operator_for_field_source = get<Tags::div<AuxiliaryField<Dim>>>(
      divergence(div_vars, mesh, inverse_jacobian)));`
I'd prefer you not make it two separate calls